solutions/cses-2079.mdx

```diff
@@ -18,40 +18,107 @@ For more information about centroids and how to find/use them, see
 #include <bits/stdc++.h>
 using namespace std;
 
-const int maxn = 2e5+5;
-int n; // number of nodes
-vector<int> g[maxn]; // graph
-int s[maxn]; // size of subtree
+const int maxn = 2e5 + 5;
+int n;				// number of nodes
+vector<int> g[maxn];  // graph
+int s[maxn];		  // size of subtree
 
 void dfs_size(int cur, int par) {
 	s[cur] = 1;
-	for(int chi : g[cur]) if(chi != par) {
-		dfs_size(chi, cur); s[cur] += s[chi];
+	for (int chi : g[cur]) {
+		if (chi != par) {
+			dfs_size(chi, cur);
+			s[cur] += s[chi];
+		}
 	}
 }
 
-int get_centroid(int cur, int sz, int par) {
-	for(int chi : g[cur]) if(chi != par) {
-		if(s[chi]*2 > sz)
-			return get_centroid(chi, sz, cur);
+int get_centroid(int cur, int par) {
+	for (int chi : g[cur]) {
+		if (chi != par) {
+			if (s[chi] * 2 > n) return get_centroid(chi, cur);
+		}
 	}
 	return cur;
 }
 
 int main() {
 	// input
 	cin >> n;
-	for(int i=1; i<n; i++) {
-		int a, b; cin >> a >> b;
+	for (int i = 1; i < n; i++) {
+		int a, b;
+		cin >> a >> b;
 		g[a].push_back(b);
 		g[b].push_back(a);
 	}
 	dfs_size(1, -1);
-	int centroid = get_centroid(1, n, -1);
+	int centroid = get_centroid(1, -1);
 	cout << centroid << '\n';
 }
 ```
 
 </CPPSection>
 
+<JavaSection>
+
+```java 
+import java.io.*;
+import java.util.*;
+
+public class Centroid {
+	private static int n; // number of nodes
+	private static List<List<Integer>> graph;
+	private static int[] sub_array_size;
+	private static Kattio io;
```

Comment:
doesn't need to be static (`io`)

Reply:
I don't think it compiles without static because the main method is static

Reply:
ill put the declaration in the main function